doc/source/whatsnew/v0.18.1.txt

```diff
@@ -22,6 +22,8 @@ New features
 Enhancements
 ~~~~~~~~~~~~
 
+Partial string matches on `DateTimeIndex` now work when part of a `MultiIndex`
+(:issue:`10331`)
 
```

Comment:
pls add a min-example here (you can add a sub-section if you'd like to highlite to setup an example and show how it works)